### PR TITLE
KEYCLOAK-7858 - OIDC servlet filter adapter OSGi support

### PR DESCRIPTION
**Pull request by dteleguin**
OSGification of Keycloak OIDC servlet filter adapter.

### CODE REVIEW

**Comment by hmlnarik** [comment]
The PR looks good to me.
@grgrzybek Could you please add your OSGi-expert review here?

**Comment by gkgeorgiev** [comment]
@dteleguin Thanks for this commit! I've been fighting with exactly the same for a while! Are you using Glassfish/Payara?

File: adapters/oidc/servlet-filter/pom.xml [hunk 1]
```diff
@@ -30,6 +30,16 @@
     <name>Keycloak Servlet Filter Adapter Integration</name>
     <description/>
 
+    <properties>
+        <keycloak.osgi.export>
+            org.keycloak.adapters.servlet.*
+        </keycloak.osgi.export>
+        <keycloak.osgi.import>
+            org.keycloak.*;version="${project.version}",
```
Comment (gkgeorgiev) [inline adapters/oidc/servlet-filter/pom.xml]: Does this mean that all other keycloak dependencies of this Adapter are also deployed as OSGI bundles?

Comment (dteleguin) [reply adapters/oidc/servlet-filter/pom.xml]: Yes, that's correct. The complete list of the deps is the following:

* keycloak-adapter-core
* keycloak-adapter-spi
* keycloak-authz-client
* keycloak-common
* keycloak-core
* keycloak-osgi-thirdparty (pre-4.2.0)
* keycloak-servlet-adapter-spi

All the modules, except for keycloak-servlet-adapter-spi, have been previously published as OSGi bundles. This PR fills the missing piece.

**Comment by dteleguin** [comment]
@gkgeorgiev, I've yet tested this on Apache Felix only. Since Glassfish/Payara employs Felix as OSGi runtime, I hope this works (however I guess the HttpService implementation would be different). Any feedback is welcome.

At the moment, the adapter needs to be registered within HttpService manually (e.g. via bundle activator). The code will land in the docs soon, so stay tuned.

File: adapters/oidc/servlet-filter/pom.xml (hunk 1 above)
Comment (hmlnarik) [inline adapters/oidc/servlet-filter/pom.xml]: Would you add explicit dependency to `javax.servlet` package similarly to [this](https://github.com/AcutusPro/keycloak/blob/42e6337954996275389cf70494a3b313816494e0/adapters/oidc/osgi-adapter/pom.xml#L40)? (Thanks @grgrzybek for pointin this out)

**Comment by hmlnarik** [comment]
@dteleguin Hi, will you have time to update the PR as per above review comment and add the documentation?

**Comment by hmlnarik** [comment]
@dteleguin yup, that would be more than welcome. Similarly to [Fuse 7/Karaf adapter](https://github.com/keycloak/keycloak-documentation/blob/master/securing_apps/topics/oidc/java/fuse7-adapter.adoc), it can be also part of the Securing applications guide, as a section next to or subsection within [servlet filter adapter](https://github.com/keycloak/keycloak-documentation/blob/master/securing_apps/topics/oidc/java/servlet-filter-adapter.adoc) documentation.

Feel free to submit any documentation you would find reasonable, our documentation expert will help with shaping it to appropriate sections/formatting/language.

**Comment by dteleguin** [comment]
Hi, hopefully will commit later today. I was considering to so the same for SAML adapter too, but finally decided to do it as a separate JIRA/PR. WDYT?

**Comment by hmlnarik** [comment]
@dteleguin Thank you
Ad SAML filter OSGi-zation - whatever way is better for you, separate JIRA/PR works for me.

**Comment by hmlnarik** [comment]
@dteleguin Thanks for the update
Could you please file a PR with documentation and link it here, and squash the commits into a single one?

**Comment by dteleguin** [comment]
@hmlnarik ok, working on the docs now
BTW my previous build has failed due to some seemingly unrelated errors - is it something on the CI side?

**Comment by hmlnarik** [comment]
Seems so, I've already restarted it

**Comment by hmlnarik** [comment]
@dteleguin Thanks for the updates. Once the docs PR would be ready, this PR can be merged.

**Comment by dteleguin** [comment]
https://github.com/keycloak/keycloak-documentation/pull/453

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**